gkShell/gk.sh

```diff
@@ -0,0 +1,507 @@
+#!/bin/bash
+# Windows requirement:
+#  - must have run dockerShell
+#    alias dockerShell='/C/Programmes/DockerToolbox/start.sh'
+#
+# Tips:
+#  - you could append to ~/.bashrc and alias to this script:
+#    alias gk="./gkShell/gk.sh"
+# Doc:
+# - docker youtube tutorials: https://plus.google.com/+XavierDurand/posts/5DGSgeMghyi
+###
+# Global parameters
+DOCKER_CMD=docker
+DOCKER_DEFAULT_MACHINE=default
+IMAGE_NAME=geokrety
+IMAGE_TAG=v0
+IMAGE_FULLNAME=${IMAGE_NAME}:${IMAGE_TAG}
+CONTAINER_NAME=devGeokrety
+DB_CONTAINER_NAME=dbGeokrety
+ADM_CONTAINER_NAME=adminer
+VOLUME_NAME=GKvolume
+
+# mount points
+SHARED_FOLDER_GKWEBSITE=gkWebsite
+SHARED_FOLDER_GKCONFIG=gkConfig
+SHARED_FOLDER_GKSCRIPTS=gkScripts
+CONTAINER_MOUNT_TARGET=/var/www/html
+CONTAINER_MOUNT_CONFIG=/var/www/config
+CONTAINER_MOUNT_SCRIPTS=/opt/geokrety-scripts
+
+###
+bold() { echo "\e[1m${*}\e[0m"; }
+posixToWindows() {
+ echo "$1" | sed 's/^\///' | sed 's/\//\\/g' | sed 's/^./\0:/'
+}
+isWindowsHost() {
+  return $( [ "$OS" == "Windows_NT" ] );
+}
+###
+# gkShell commands
+VOLUME_COMMANDS="volume-create|volume-ls|volume-rm|volume-inspect"
+if isWindowsHost; then
+  INSTALL_COMMANDS="composer|install|uninstall"
+  ALLOWED_COMMANDS="help|info|${INSTALL_COMMANDS}|build|images|run|ps|start|stop|rm|sh|logs|tail|alias|dm-ip|dm-sh"
+else
+  ALLOWED_COMMANDS="help|info|composer|build|images|run|ps|start|stop|rm|sh|logs|tail|alias"
+  DOCKER_CMD="sudo docker"
+fi
+# don't tell about volume for now
+EXHAUSTIVE_LIST_OF_COMMANDS="${ALLOWED_COMMANDS}|${VOLUME_COMMANDS}"
+#
+#
+dockerNeeded() {
+	echo " X Docker is required : 'docker version' must success"
+	echo
+	echo " - windows users :"
+	echo "   install msi file from https://docs.docker.com/toolbox/toolbox_install_windows/ (recommended),"
+    echo "     or use PowerShell admin : choco install command-toolbox".
+    echo "   then start dockerShell : /C/Programmes/DockerToolbox/start.sh"
+    echo
+	echo " - linux users :"
+	echo "   follow docker install pages : https://docs.docker.com/install/linux/command-ce/ubuntu/"
+	exit 1
+}
+checkDocker() {
+   ${DOCKER_CMD} version 2>/dev/null 1>/dev/null|| dockerNeeded
+}
+
+
+# Computed parameters
+computedParams() {
+  PROJECT_DIR="$(cd -P -- "$(dirname -- "$0")" && cd .. && pwd -P)"
+  WEBSITE_DIR="${PROJECT_DIR}/website"
+  CONFIG_DIR="${PROJECT_DIR}/gkShell/config"
+  SCRIPTS_DIR="$(cd -P -- "${PROJECT_DIR}/../geokrety-scripts/" 2>/dev/null && pwd -P)"
```

Comment:
Don't remember to have seen this requirement in the INSTALL docs :P

Reply:
yep TODO

Reply:
added